src/content/tutorials/github-twilio-notifications/index.md

```diff
@@ -0,0 +1,287 @@
+}
+```
+
+Let's start and refactor the starter code to handle a POST response and rename the request handler. We can use the `request.method` property of [request](link to request) to check if the request is a `POST` request and send an error response if incorrect. The `simpleResponse` fucntion is an easy wrapper for you to send requests with your Worker.
+
+```javascript
+---
+filename: index.js
+---
+function simpleResponse(statusCode, message) {
+  let resp = {
+    message: message,
+    status: statusCode
+  }
+
+  return new Response(JSON.stringify(resp), {
+    headers: new Headers([["Content-Type", "application/json"]]),
+    status: statusCode
+  })
+}
+
+addEventListener("fetch", event => {
+  event.respondWith(githubWebhookHandler(event.request))
+})
+
+async function githubWebhookHandler(request) {
+  if (request.method !== "POST") {
+    return simpleResponse(
+      200,
+      `Please send a POST request :)`
+    )
+  }
+}
+```
+
+The next challenge is to check validate the request is sent with the right secret key. GitHub attaches a hash signature for each payload using the secret key. [Github's Documentation](https://developer.github.com/webhooks/securing/) We can use a helper function called checkSignature on the request the hash is correct. Then, we can access data from the webhook by parsing the request as json.
+
+```javascript
+---
+filename: index.js - githubWebhookHandler()
+---
+async function githubWebhookHandler(request) {
+  if (request.method !== "POST") {
+    return simpleResponse(
+      200,
+      `Please send a POST request :)`
+    )
+  }
+  try{
+    if (!checkSignature(request)) {
+      return simpleResponse(403, "Incorrect Secret Code")
+    }
+
+    const formData = await request.json()
+    const repo_name = formData.repository.full_name
+    const action = await request.headers.get("X-GitHub-Event")
+    const sender_name = formData.sender.login
+  }
+  catch (e) {
+    return simpleResponse(
+      200,
+      `Error:  ${e} `
+    )
+  }
+}
+```
+
+The checkSignature function will use the crypto library to hash the received payload with our known secret key to ensure it matches the request hash. GitHub uses a HMAC hexdigest to compute the hash in the sha1 format [Github's Documentation](https://developer.github.com/webhooks/securing/).
+
+```javascript
+---
+filename: index.js
+---
+
+const crypto = require('crypto')
+
+async function createHexSignature(requestBody) {
+  let hmac = crypto.createHmac('sha1', SECRET_TOKEN)
+  hmac.update(requestBody,'utf-8')
+
+  return hmac.digest('hex')
+}
+
+async function checkSignature(request) {
+  let expectedSignature = await createHexSignature(await request.text())
+  let actualSignature = await request.headers.get("X-Hub-Signature")
+
+  return expectedSignature === actualSignature
+}
+```
+
+Since our project relies on importing a library, we need to also update our `wrangler.toml` and set `type = "webpack"`.
```

Comment:
should have a code sample here, and i'd also considering moving this up into the prereqs/setup section, but up to you on that one